pkgs/by-name/ca/caddy/plugins.nix

```diff
@@ -7,19 +8,44 @@
   git,
   caddy,
 }:
+
+let
+  inherit (builtins) hashString;
+  inherit (lib)
+    assertMsg
+    concatMapStrings
+    elemAt
+    fakeHash
+    filter
+    hasInfix
+    length
+    lessThan
+    sort
+    toShellVar
+    ;
```

Comment:
Out of curiosity, is that a best practice? Is it explained somewhere?

Reply:
Neither. I particularly don't like but it doesn't hurt (as long you're not using `with lib`).

Reply:
I've seen a few modules (in nixpkgs of course) pull in library functions right after getting arguments like these:
- https://github.com/NixOS/nixpkgs/blob/821e42c405b7ff755b0a32f5a5078357fed45354/nixos/modules/services/monitoring/metricbeat.nix#L8-L17
- https://github.com/NixOS/nixpkgs/blob/4506ece030a0c82d078edd0360ea8af6b4d94035/nixos/modules/services/networking/mtr-exporter.nix#L8-L23
- https://github.com/NixOS/nixpkgs/blob/4506ece030a0c82d078edd0360ea8af6b4d94035/nixos/modules/services/networking/ergo.nix#L13-L20
And I've taken quite a liking to this style, so yeah... Not sure if this is the "best practice" tho